gtsfm/pose_slam/pose_slam.py

```diff
@@ -20,6 +23,100 @@
 
 
 class PoseSlam:
+    """Initializes using PoseSLAM given relative pose priors."""
+
+    @staticmethod
+    def angle(R1, R2):
+        """Calculate angle between two rotations, in degrees."""
+        return np.degrees(np.linalg.norm(R1.logmap(R2)))
+
+    @staticmethod
+    def difference(P1, P2):
+        """Calculate the translation and angle differences of two poses.
+        P1, P2: Pose3
+        Return:
+            distance: translation difference
+            angle: angular difference
+        """
+        # TODO(frank): clean this up
+        t1 = P1.translation()
+        t2 = P2.translation()
+        R1 = P1.rotation()
+        R2 = P2.rotation()
+        R1_2 = R1.compose(R2.inverse())
+        t1_ = R1_2.rotate(t2)
+        # t1_2 = t1 - R1_2*t2
+        distance = np.linalg.norm(t1 - t1_)
+        angle_ = PoseSlam.angle(R1, R2)
+        return distance, angle_
+
+    @staticmethod
+    def filter_constraints(constraints: List[Constraint], poses: List[Pose3]) -> List[Constraint]:
+        """Filter constraints according to notebook criteria.
+
+        Args:
+            constraints (List[Constraint]): relative constraints from file
+            poses (List[Pose3]): estimated initial values
+
+        Returns:
+            Filtered constraints.
+        """
+
+        filtered_constraints: List[Constraint] = []
+
+        # Filter on covariance and error
+        for constraint in constraints:
+            a, b = constraint.a, constraint.b
+            aTb, cov = constraint.aTb, constraint.cov
+            predicted_aTb = poses[a].between(poses[b])
+            trans_diff, rot_diff = PoseSlam.difference(aTb, predicted_aTb)
+            inlier = (trans_diff <= 0.04) and (rot_diff <= 5)
```

Comment:
Just 4cm? Shicong was giving an example of a dataset where he observed the start and end point of the trajectory to be 2 metres apart and he wanted to get loop closure there?

Reply:
is this a check for very high values ?

Reply:
thought we had another function to remove those.

Reply:
I agree with both comments and will revert this method to just check covariances for now, so we can merge this.